sdk/client/src/blocking.rs

```diff
@@ -57,22 +57,30 @@ impl BlockingClient {
         timeout: Option<Duration>,
         delay: Option<Duration>,
     ) -> Result<Response<TransactionView>, WaitForTransactionError> {
-        self.wait_for_transaction(
+        let response = self.wait_for_transaction(
             txn.sender(),
             txn.sequence_number(),
             txn.expiration_timestamp_secs(),
-            &Transaction::UserTransaction(txn.clone()).hash().to_hex(),
+            Transaction::UserTransaction(txn.clone()).hash(),
             timeout,
             delay,
-        )
+        )?;
+
+        if !response.inner().vm_status.is_executed() {
+            return Err(WaitForTransactionError::TransactionExecutionFailed(
+                response.into_inner(),
+            ));
+        }
+
+        Ok(response)
     }
 
     pub fn wait_for_transaction(
         &self,
         address: AccountAddress,
         seq: u64,
         expiration_time_secs: u64,
-        txn_hash: &str,
+        txn_hash: HashValue,
         timeout: Option<Duration>,
         delay: Option<Duration>,
     ) -> Result<Response<TransactionView>, WaitForTransactionError> {
```

Comment:
should we change this `WaitForTransactionError` to not include `TransactionExecutionFailed`? and use different error type that includes `TransactionExecutionFailed` for above `wait_for_signed_transaction` function?

Reply:
I don't think thats strictly necessary as its ok for a function to return an Error type where it won't construct all the different variants.